api/filter/http_fault.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "api/rds.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+// Delay specification is used to inject latency into the rpc operations.
+message Delay {
+  // Delay type to use (fixed|exponential|..). Currently, only fixed delay (step function) is supported.
+  string type = 1;
+
+  // An integer between 0-100 indicating the percentage of operations
+  // on which the delay will be injected.
+  google.protobuf.UInt32Value percent = 2;
+
+  oneof http_delay_type {
+    // Add a fixed delay before forwarding the operation upstream. Format: 1h/1m/1s/1ms. MUST be >=1ms.
+    google.protobuf.Duration fixed_delay = 2;
```

Comment:
This has  field number 2, but so does `percent` above. `oneof` fields share the same number space as the fields in the message around them.